### PR TITLE
persist new historical metrics to sql

### PR DESCRIPTION
**Pull request by neonphog**
DEPENDS ON https://github.com/holochain/holochain/pull/1182

### Summary

- persist historical metrics to metrics database once per hour

### TODO:
- [x] CHANGELOG(s) updated with appropriate info
- [ ] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

### CODE REVIEW

File: crates/holochain/CHANGELOG.md [hunk 1]
```diff
@@ -5,8 +5,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 ## Unreleased
 
 - Fixes issue where holochain could get stuck in infinite loop when trying to send validation receipts. [#1181](https://github.com/holochain/holochain/pull/1181).
-
 - Additional networking metric collection and associated admin api `DumpNetworkMetrics { dna_hash: Option<DnaHash> }` for inspection of metrics [#1160](https://github.com/holochain/holochain/pull/1160)
+- **BREAKING CHANGE** - Schema change for metrics database. Holochain will persist historical metrics once per hour, if you do not clear the metrics database it will crash at that point. [#1183](https://github.com/holochain/holochain/pull/1183)
```
Comment (freesig) [inline crates/holochain/CHANGELOG.md:9]: Does this just mean it needs to be cleared once if you have an existing database. Also maybe we could provide some instructions on how to clear the db?

**Review by freesig — APPROVED** [review]

File: crates/holochain_sqlite/src/sql/p2p_metrics/insert.sql [hunk 2]
```diff
@@ -0,0 +1,16 @@
+INSERT
+  OR IGNORE INTO p2p_metrics (
```
[the remark this answers is not in the document]
Comment (neonphog) [reply crates/holochain_sqlite/src/sql/p2p_metrics/insert.sql:2]: the only constraints are not nulls, or the unique on the rowid... basically if you have a mal-formed statement... i figured ignoring garbage was a good idea

File: crates/holochain/CHANGELOG.md (hunk 1 above)
Comment (neonphog) [reply crates/holochain/CHANGELOG.md:9]: yep, cleared. feel free to edit my message, I don't know how to clear the db : )

**Review by thedavidmeister — APPROVED** [review]

File: crates/holochain_sqlite/src/sql/p2p_metrics/insert.sql (hunk 2 above)
[the remark this answers is not in the document]
Comment (neonphog) [reply crates/holochain_sqlite/src/sql/p2p_metrics/insert.sql:2]: gonna go ahead and merge this, feel free to edit the changelog post facto

**Pull request merged**